src/lib/core/data-table/data-table.spec.ts

```diff
@@ -336,6 +426,44 @@ class CustomRoleCdkTableApp {
   @ViewChild(CdkTable) table: CdkTable<TestData>;
 }
 
+@Component({
+  template: `
+    <cdk-table [dataSource]="dataSource">
+      <ng-container cdkColumnDef="column_a">
+        <cdk-header-cell *cdkHeaderCellDef> Column A</cdk-header-cell>
+        <cdk-cell *cdkCellDef="let row; let first = first;
+                               let last = last; let even = even; let odd = odd"
+                  [ngClass]="{
+                    'custom-cell-class-first': enableCellContextClasses && first,
+                    'custom-cell-class-last': enableCellContextClasses && last,
+                    'custom-cell-class-even': enableCellContextClasses && even,
+                    'custom-cell-class-odd': enableCellContextClasses && odd
+                  }">
+          {{row.a}}
+        </cdk-cell>
+      </ng-container>
+      <cdk-header-row *cdkHeaderRowDef="columnsToRender"></cdk-header-row>
+      <cdk-row *cdkRowDef="let row; columns: columnsToRender;
+                           let first = first; let last = last; let even = even; let odd = odd"
+               [ngClass]="{
+                 'custom-row-class-first': enableRowContextClasses && first,
+                 'custom-row-class-last': enableRowContextClasses && last,
+                 'custom-row-class-even': enableRowContextClasses && even,
+                 'custom-row-class-odd': enableRowContextClasses && odd
+               }">
+      </cdk-row>
+    </cdk-table>
+  `
+})
+class RowContextCdkTableApp {
+  dataSource: FakeDataSource = new FakeDataSource();
+  columnsToRender = ['column_a'];
+  enableRowContextClasses = false;
+  enableCellContextClasses = false;
+
+  @ViewChild(CdkTable) table: CdkTable<TestData>;
```

Comment:
Whoops, either a copy-paste problem or eventually didn't need the table in the tests. Thanks!